app/models/course_user.rb

```diff
@@ -1,5 +1,32 @@
 class CourseUser < ActiveRecord::Base
+  stampable
   belongs_to :user, inverse_of: :course_users
   belongs_to :course, inverse_of: :course_users
-  enum role: { student: 0, teaching_assistant: 1, manager: 2, owner: 3 }
+
+  enum role: { student: 0, teaching_assistant: 1, manager: 2, owner: 3 },
+       status: { pending: 0, approved: 1 }
+
+  scope :staff, -> { where(role: [roles[:teaching_assistant], roles[:manager], roles[:owner]]) }
+
+  def staff?
+    teaching_assistant? || manager? || owner?
+  end
+
+  def staff_of_the_course
+    course.course_users.staff
+  end
+
+  # Approve the registration record by creating a CourseUser of the user to the course with the
+  # requested role.
+  def approve
+    update(status: CourseUser.statuses[:approved])
```

Comment:
Yup, initially I passed in a Symbol and it complained. That's why I changed to refer to the end.